service/rpc2/server.go

```diff
@@ -132,9 +133,49 @@ func (s *RPCServer) Command(command api.DebuggerCommand, cb service.RPCCallback)
 		cb.Return(nil, err)
 		return
 	}
+
 	var out CommandOut
 	out.State = *st
-	cb.Return(out, nil)
+	if clientIsConnected := cb.Return(out, nil); s.config.Headless && !clientIsConnected {
```

Comment:
We can't do this in this layer. It will not work if delve is started with `dlv dap` or if someone connects to it using VSCode (or any other program that uses the DAP protocol) and then that client quits.

Reply:
Right, I missed that.  I'm leaning towards doing this in both `rpc2` and `dap` or do you suggest that we should do it in the `debugger` layer instead?

Reply:
I lean slightly towards doing it in debugger but both would be ok.

Reply:
Hi @aarzilli , correct me if i'm wrong, after checking dap's code I'm not sure if we need to implement this for dap. reasons being:
1. `dap` doesn't support multiclient
2. `dap` will quit when the client disconnected (so printing the log here won't really make sense)

ref:
```
The --client-addr flag is a special flag that makes the server initiate a debug session
by dialing in to the host:port where a DAP client is waiting. This server process
will exit when the debug session ends.
```

Reply:
You might be right about `dlv dap` but DAP connections are also accepted when delve is started with `dlv --headless debug`. See serveConnectionDemux in service/rpccommon/server.go as well as onDisconnectRequest in service/dap/server.go.

Reply:
Ah yeah you're right, thanks for the clarification.